wolfcrypt/src/cmac.c

```diff
@@ -230,9 +230,16 @@ int wc_CmacFinal(Cmac* cmac, byte* out, word32* outSz)
     const byte* subKey;
     word32 remainder;
 
-    if (cmac == NULL || out == NULL || outSz == NULL) {
+    if (cmac == NULL)
         return BAD_FUNC_ARG;
+
+    if (out == NULL || outSz == NULL) {
+        if ((out == NULL) ^ (outSz == NULL))
```

Comment:
Use '&&' as it is clearer.

Reply:
that whole thing was a mess.  gone with the refactor I just pushed.